geomstats/geometry/stratified/point_set.py

```diff
@@ -1,91 +1,61 @@
 """Class for Stratified Spaces.
 
-Lead authors: Anna Calissano & Jonas Lueg
+Lead authors: Anna Calissano and Jonas Lueg
 """
 
-import functools
-import itertools
 from abc import ABC, abstractmethod
 
+import geomstats.backend as gs
 
-def broadcast_lists(list_a, list_b):
-    """Broadcast two lists.
 
-    Similar behavior as ``gs.broadcast_arrays``, but for lists.
-    """
-    n_a = len(list_a)
-    n_b = len(list_b)
-
-    if n_a == n_b:
-        return list_a, list_b
-
-    if n_a == 1:
-        return itertools.zip_longest(list_a, list_b, fillvalue=list_a[0])
-
-    if n_b == 1:
-        return itertools.zip_longest(list_a, list_b, fillvalue=list_b[0])
-
-    raise Exception(f"Cannot broadcast lens {n_a} and {n_b}")
-
-
-def _manipulate_input(arg):
-    if type(arg) not in [list, tuple]:
-        return [arg]
-
-    return arg
-
-
-def _vectorize_point(*args_positions, manipulate_input=_manipulate_input):
-    """Check point type and transform in iterable if not the case.
-
-    Parameters
-    ----------
-    args_positions : tuple
-        Position and corresponding argument name. A tuple for each position.
-
-    Notes
-    -----
-    Explicitly defining args_positions and args names ensures it works for all
-    combinations of input calling.
-    """
-
-    def _dec(func):
-        @functools.wraps(func)
-        def _wrapped(*args, **kwargs):
-            args = list(args)
-            for pos, name in args_positions:
-                if name in kwargs:
-                    kwargs[name] = manipulate_input(kwargs[name])
-                else:
-                    args[pos] = manipulate_input(args[pos])
-
-            return func(*args, **kwargs)
+class Point(ABC):
+    """Class for points of a set."""
 
-        return _wrapped
+    @abstractmethod
+    def equal(self, point, atol=gs.atol):
+        """Check equality against another point.
 
-    return _dec
+        Parameters
+        ----------
+        point : Point or list[Point]
+            Point to compare against point.
+        atol : float
 
+        Returns
+        -------
+        is_equal : array-like, shape=[...]
+        """
 
-class Point(ABC):
-    r"""Class for points of a set."""
 
-    @abstractmethod
-    def __repr__(self):
-        """Produce a string with a verbal description of the point."""
+class PointCollection(ABC, list):
```

Comment:
Explain what this is, explain that this is the analogous of a batch.
Maybe rename it PointBatch?